### ISSUE TITLE
Mixing Hash and keyword arguments raises an error

### ISSUE THREAD

**Issue by robin850** [jruby/jruby#3138]
Hello,

There's a bug with JRuby 9000 when defining a method that mixes a Hash argument with a keyword argument:

~~~ruby
>> def foo(bar = {}, baz: nil)
 | end
=> :foo
>> foo({ "foo" => "bar" })
=> ArgumentError: unknown keyword: foo
~~~

This is not a problem with MRI.

Have a nice day!

**Comment by enebo** [comment]
Yeah.  This is something we know is broken and we discovered it pretty late in development.  We will be targeting this in an earlier point release of 9.0.x.y.  There are also tagged out specs as well so we have coverage once we do fix this.

**Comment by headius** [comment]
Still broken as of master today (9.0.5.0).

**Comment by headius** [comment]
Still failing today, but it won't make 9.1.

**Comment by enebo** [comment]
Fixed in commit c18394331345f650a49f8d47b5e4c404788c3b62
